### PR TITLE
ci: check for mayastor pod restarts when running e2e-tests CAS-612

### PR DESCRIPTION
**Pull request by blaisedias**
Add script which checks for mayastor pod restarts,
invoke it after every test run

### CODE REVIEW

**Review by chriswldenyer — APPROVED** [review]
LGTM

File: scripts/e2e_check_pod_restarts.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```
Comment (jonathan-teh) [inline scripts/e2e_check_pod_restarts.sh:1]: /usr/bin/env for portability, and consistency with other scripts.

File: scripts/e2e_check_pod_restarts.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+set -e
+
+restarts=$(kubectl get pods -n mayastor | grep -e mayastor -e moac | awk '{print $4}')
+for num in $restarts
+do
+    if [ "$num" -ne "0" ]
+    then
```
Comment (jonathan-teh) [inline scripts/e2e_check_pod_restarts.sh:9]: Looks more consistent with 'if' and 'then' on the same line, as in the other script above.

File: scripts/e2e_check_pod_restarts.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+set -e
+
+restarts=$(kubectl get pods -n mayastor | grep -e mayastor -e moac | awk '{print $4}')
```
Comment (jonathan-teh) [inline scripts/e2e_check_pod_restarts.sh:5]: An example of what the grep matches would be useful.

**Review by jonathan-teh — APPROVED** [review]
Looks fine, just nitpicks.

File: scripts/e2e_check_pod_restarts.sh (hunk 1 above)
Comment (blaisedias) [reply scripts/e2e_check_pod_restarts.sh]: Done

File: scripts/e2e_check_pod_restarts.sh (hunk 2 above)
Comment (blaisedias) [reply scripts/e2e_check_pod_restarts.sh]: Done

File: scripts/e2e_check_pod_restarts.sh (hunk 3 above)
Comment (blaisedias) [reply scripts/e2e_check_pod_restarts.sh:12]: Done

**Comment by blaisedias** [comment]
bors merge

**Pull request merged**